cmd/kops/toolbox_template.go

```diff
@@ -166,14 +163,39 @@ func runToolBoxTemplate(f *util.Factory, out io.Writer, options *toolboxTemplate
 		if err != nil {
 			return fmt.Errorf("unable to render template: %s, error: %s", x, err)
 		}
-
+		if len(rendered) <= 0 {
+			continue
+		}
 		io.WriteString(writer, rendered)
 
 		// @check if we should need to add document separator
 		if i < size {
 			io.WriteString(writer, "---\n")
 		}
 	}
+	content := writer.Bytes()
+
+	if options.formatYAML {
+		var data interface{}
+		err := yaml.Unmarshal(content, &data)
```

Comment:
Is `---` actually part of yaml syntax?  i.e. does this work with more than one template?

If not, we might have to format each section independently..